### PR TITLE
Handle logging in async function

### PR DESCRIPTION
**Pull request by rynop**
- [x] This is a small change 
---

**Description:**

Sorry I didn't file an issue cuz this was such a small change. 

When using mocha, and trying to log within a hook/test `detox/src/utils/customConsoleLogger.js:: getOrigin()` blows up because `userCallsite.getFileName()` is `undefined` and therefore `path.relative()` explodes.

Simple reproduce:
```
  before: async () => {
    console.log('**********BLAH');
```

PR is not the most elegant solution, but not sure there is much that can be done when using https://github.com/sindresorhus/callsites

Without this fix, I can not use Detox (and I want to, cuz really like it :)

### CODE REVIEW

File: detox/src/utils/customConsoleLogger.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ function getStackDump() {
 
 function getOrigin() {
   const userCallsite = callsites()[USER_STACK_FRAME_INDEX];
-  const filename = path.relative(process.cwd(), userCallsite.getFileName());
+  const filename = userCallsite && userCallsite.getFileName() ? path.relative(process.cwd(), userCallsite.getFileName()) : '?';
```
Comment (noomorph) [inline detox/src/utils/customConsoleLogger.js:14]: Could you split this line into two with
```js
const callsiteFilename = userCallsite && userCallsite.getFileName();`
```
beforehand?

**Comment by d4vidi** [comment]
I must emphasize what @noomorph is asking regarding tests. This is TDD code -- need to add a failing test that acknowledges this specific case.

Also, please don't use `?` for the file name, but an `<unknown>` notation.

Other than that: thank you for this! 😄

File: detox/src/utils/customConsoleLogger.js (hunk 1 above)
Comment (rynop) [reply detox/src/utils/customConsoleLogger.js]: For sure, NP.

**Comment by rynop** [comment]
Fixed up and test added!

File: detox/src/utils/customConsoleLogger.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ function getStackDump() {
 
 function getOrigin() {
   const userCallsite = callsites()[USER_STACK_FRAME_INDEX];
-  const filename = path.relative(process.cwd(), userCallsite.getFileName());
+  const callsiteFilename = userCallsite && userCallsite.getFileName();
+  const filename = callsiteFilename ? path.relative(process.cwd(), callsiteFilename) : '<unknown>';
   return `at ${filename}:${userCallsite.getLineNumber() || '?'}:${userCallsite.getColumnNumber() || '?'}`;
```
Comment (noomorph) [inline detox/src/utils/customConsoleLogger.js:16]: I am afraid that in case when userCallsite is indeed falsy, the attempt to get line and column number will fail as well. If you have started already guarding against this edge case, make sure to make it consistent here too.

Comment (rynop) [reply detox/src/utils/customConsoleLogger.js]: It didn't in my use case (I was surprised as well). I'll add logic tho.

Leaving test case as is cuz the line/col is obv non-deterministic (given my use case findings).  The update code will handle both scenarios.

**Comment by noomorph** [comment]
@rynop, I am sorry for the delay - the last update from you somehow slipped between the cracks, and I have not noticed it. I'll review your PR soon.

**Comment by rynop** [comment]
@noomorph no worries!

**Comment by noomorph** [comment]
@rynop, thank you! Great job! :+1:
